### DISCUSSION TITLE
Archive extension

### DISCUSSION THREAD

**Discussion by kmodali** [stac-utils/.github#7] [Q&A]
Dear All, 
     I have _'*.tar'_ files as assets and hence using the **archive extension** specification to create my catalog. 
Subsequently I added the **archive extension**  implementation in _'pystac'_ and _'xpystac'_ . Then I could convert the _'*.tar'_ file from the asset object into _'xarray'_ dataset , as below:
`archive_asset=item.assets['tape1']`
` archive_asset.to_dict() `
_{'href': 'test_1.tar',
 'type': 'application/x-tar',
 'archive:format': 'application/x-tar',
 'archive:href': 'exp_1.zarr/',
 'archive:type': 'application/vnd+zarr'}_

`ds_tar=xr.open_dataset(archive_asset,engine="stac")`

Now, I have the scenario where the catalog search, results in a list of items and each item has _'*.tar'_ file as asset. I would be thankful if someone could clarify the **archive extension** application in such scenario. Any other idea(s), suggestion(s) in this direction is kindly solicited.
Thank you.

**Comment by jsignell** [comment]
`xpystac` relies on the `type` to determine which xarray backend to use. So it doesn't know what to do with 'application/x-tar'. Do you know if xarray has a reader backend for archived zarr files? If so then it would just be a few lines of code to map from the `type` to the corresponding backend.

Stepping back a bit, it seems like an anti-pattern to create archives of zarr files. The more cloud-friendly approach would be to enable per-chunk compression.

**Reply by kmodali** [reply]
@jsignell  Thank you very much for the clarification. Yes, i am using the  `type`  from the asset and using `tarfile` module to extract the file from the tar archive , as i am not aware of 'archived zarr' backend. Could you please send a link to an example usage. 
As tape storage is (still) part of the storage infrastructure, this necessitates the creation of archives(tar) of zarr files. 
In between i came to know about the `ZipStore` as an alternative and it's cloud friendly approach.

**Reply by jsignell** [reply]
Ah so you are essentially using a custom backend reader (first you extract the zarr file, then you read it). I guess in theory you could create a custom reader (docs on that: https://docs.xarray.dev/en/latest/internals/how-to-add-new-backend.html) and then extend xpystac to pass off to that custom reader if the `type` is 'application/x-tar'. You would put that right here: https://github.com/stac-utils/xpystac/blob/7b77eccc08533739ca9631826ad369331df02ca5/xpystac/core.py#L126C1-L128C62

xpystac isn't really set up to easily be extended in that way. So you are probably better off just using the underlying packages (stackstac or odc-stac, xarray, pystac) if you cannot switch from having the entire zarr file archived to having chunk-level compression.
